### PR TITLE
refactored code structure

### PR DESCRIPTION
**Pull request by leepc12**
No new feature is added to this PR. This PR includes 1) refactoring (mostly moving code lines to a correct place) and 2) adding logger to each module.

Code refactoring
- Ordered imports
- Separated `caper/caper.py`'s CLI into `caper/cli.py`.
- Modified executable `bin/caper` to point to a correct CLI `main()`.
- Added `caper/__main__.py` for `python -m`.
- Added `caper/version.py` to prevent circular imports

Logging
- Removed `--verbose` since it already prints INFO level logs. We still have `--debug`.
- Individual logger for each module

Future work
- Caper still has two modules (`caper/caper_args.py` and `caper/cli.py`) for CLI. Caper can parse parameters from both CLI and conf file, which makes it difficult to separate those two parts in `caper/caper_args.py`. This is not done in this RP but will be definitely done in another ticket.

### CODE REVIEW

File: caper/cromwell_rest_api.py [hunk 1]
```diff
@@ -263,17 +255,17 @@ def __request_get(self, endpoint, params=None):
                 url, auth=self._auth, params=params,
                 headers={'accept': 'application/json'})
         except Exception as e:
-            # traceback.print_exc()
-            print(e)
-            print('Help: cannot connect to server. '\
+            logger.error(e)
+            logger.error('Help: cannot connect to server. '\
                   'Check if server is dead or still spinning up.')
```
Comment (p-sud) [inline caper/cromwell_rest_api.py:260]: You can consolidate this into a single call:
```suggestion
            logger.exception('Help: cannot connect to server. '\
                  'Check if server is dead or still spinning up.')
```

File: caper/version.py [hunk 2]
```diff
@@ -0,0 +1 @@
+version = '0.8.2'
```
[the remark this answers is not in the document]
Comment (leepc12) [reply caper/version.py:1]: `caper_args.py` needs access to it to print out version `caper --version`.

I actually tried your suggestion and had `from . import __version__ as version` in `caper_args.py` due to circular import.

**Comment by p-sud** [comment]
Somehow got an email you had a comment about the version business, but I have no idea where that comment is. Your rationale makes sense though. It only works if you don't use relative imports in the module, you'd need to refactor somewhat to get reading from __init__.py to work.

**Comment by leepc12** [comment]
Sorry, I just made a comment but deleted.
I actually tried your method once but failed due to circular import.
I fixed that circular import and now it works fine.

Jin

File: caper/version.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (leepc12) [reply caper/version.py]: Thanks, fixed.

File: caper/cli.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    elif action == 'metadata':
+        c.metadata()
+    elif action == 'unhold':
+        c.unhold()
+    elif action in ['troubleshoot', 'debug']:
+        c.troubleshoot()
+    else:
+        raise ValueError(
+        	'Unsupported or unspecified action. act={a}'.format(a=action))
+    return 0
```
[the remark this answers is not in the document]
Comment (leepc12) [reply caper/cli.py]: Thanks, fixed. I chose b).

**Comment by p-sud** [comment]
Looks good now.

**Pull request merged**